packages/ivass-certified-attributes-importer/src/service/readModelQueriesServiceSQL.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  Attribute,
+  Tenant,
+  TenantId,
+  unsafeBrandId,
+  WithMetadata,
+} from "pagopa-interop-models";
+import {
+  aggregateTenantArray,
+  AttributeReadModelService,
+  TenantReadModelService,
+  toTenantAggregatorArray,
+} from "pagopa-interop-readmodel";
+import { and, eq, inArray, isNull } from "drizzle-orm";
+import {
+  attributeInReadmodelAttribute,
+  DrizzleReturnType,
+  tenantCertifiedAttributeInReadmodelTenant,
+  tenantFeatureInReadmodelTenant,
+  tenantInReadmodelTenant,
+} from "pagopa-interop-readmodel-models";
+import { IvassReadModelTenant } from "../model/tenant.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function readModelQueriesBuilderSQL(
+  readModelDB: DrizzleReturnType,
+  tenantReadModelService: TenantReadModelService,
+  attributeReadModelService: AttributeReadModelService
+) {
+  return {
+    /**
+     * Retrieve tenants that match the given tax codes, with their unrevoked certified attribute
+     */
+    async getIVASSTenants(
+      externalId: string[]
+    ): Promise<IvassReadModelTenant[]> {
+      const queryResult = await readModelDB
+        .select({
+          tenant: tenantInReadmodelTenant,
+          certifiedAttribute: tenantCertifiedAttributeInReadmodelTenant,
+          feature: tenantFeatureInReadmodelTenant,
+        })
+        .from(tenantInReadmodelTenant)
+        .innerJoin(
+          tenantCertifiedAttributeInReadmodelTenant,
+          and(
+            eq(
+              tenantInReadmodelTenant.id,
+              tenantCertifiedAttributeInReadmodelTenant.tenantId
+            ),
+            isNull(
+              tenantCertifiedAttributeInReadmodelTenant.revocationTimestamp
+            )
+          )
+        )
+        .leftJoin(
+          tenantFeatureInReadmodelTenant,
+          eq(
+            tenantInReadmodelTenant.id,
+            tenantFeatureInReadmodelTenant.tenantId
+          )
+        )
+        .where(
+          and(
+            eq(tenantInReadmodelTenant.externalIdOrigin, "IVASS"),
```

Comment:
Minor, since we're here, we could move this into a constant reused everywhere in this package

```suggestion
            eq(tenantInReadmodelTenant.externalIdOrigin, IVASS_ORIGIN_OR_A_BETTER_NAME),
```

Reply:
👍